components/KiboAddressForm.vue

```diff
@@ -137,9 +177,88 @@ export default {
       })
     }
 
+    const errors = ref({
+      firstName: "",
+      lastNameOrSurname: "",
+      address1: "",
+      address2: "",
+      cityOrTown: "",
+      stateOrProvince: "",
+      postalOrZipCode: "",
+      countryCode: "",
+      phoneNumbers: "",
+    })
+
+    const assignErrors = (errorField = "", errorMessage = "") => {
+      if (errorField) {
+        errors.value[errorField] = errorMessage
+      } else {
+        Object.keys(errors.value).filter((field) => (errors.value[field] = errorMessage))
+      }
+    }
+    const validateForm = () => {
+      schema
+        .validate(props.value)
+        .then(() => {
+          context.emit("validateForm", true)
+        })
+        .catch(() => {
+          context.emit("validateForm", false)
+        })
+    }
+    const schema = yup.object({
+      firstName: yup.string().required(context.root.$t("Required")),
+      lastNameOrSurname: yup.string().required(context.root.$t("Required")),
+      address: yup.object({
+        address1: yup.string().required(context.root.$t("Required")),
+        address2: yup.string().required(context.root.$t("Required")),
+        cityOrTown: yup.string().required(context.root.$t("Required")),
+        stateOrProvince: yup.string().required(context.root.$t("Required")),
+        postalOrZipCode: yup
+          .string()
+          .required(context.root.$t("Required"))
+          .matches(/^[0-9]+$/, "Must be only digits"),
+        countryCode: yup.string().required(context.root.$t("Required")),
+      }),
+      phoneNumbers: yup.object({
+        home: yup
+          .string()
+          .required(context.root.$t("Required"))
+          .matches(/^[+]*[(]{0,1}[0-9]{1,3}[)]{0,1}[-\s\\./0-9]*$/g, "Must be valid phone number"),
```

Comment:
Can't we move validations on useUiValidation.ts to keep all validations on one place?

Reply:
Resolved.